include/circt/Dialect/SV/SVInOutOps.td

```diff
@@ -137,3 +137,20 @@ def ArrayIndexInOutOp
   let assemblyFormat =
     "$input`[`$index`]` attr-dict `:` type($input) `,` type($index)";
 }
+
+def PartSelectOp : SVOp<"part_select", [NoSideEffect]> {
```

Comment:
This is the inout flavor, for consistency, name it PartSelectInOutOp.  Also add a non-inout flavor for rhs.

Reply:
Going to add the RHS version.